src/Core/Engine/Physics/PhysicsWorld.cpp

```
@@ -0,0 +1,203 @@
+
+    glm::vec3 centerA = transformA.LocalPosition + colliderA.centerOffset;
+    glm::vec3 halfA = colliderA.halfExtents * transformA.LocalScale;
+
+    glm::vec3 centerB = transformB.LocalPosition + colliderB.centerOffset;
+    glm::vec3 halfB = colliderB.halfExtents * transformB.LocalScale;
+
+    glm::vec3 delta = centerB - centerA;
+
+    glm::vec3 overlap = halfA + halfB - glm::abs(delta);
+
+    float penetration = overlap.x;
```

Clang (14.0.6)

[warning] variable 'penetration' is not initialized

(cppcoreguidelines-init-variables)